rbac/management/group/model.py

```diff
@@ -49,6 +52,29 @@ class Group(TenantAwareModel):
     system = models.BooleanField(default=False)
     admin_default = models.BooleanField(default=False)
 
+    @staticmethod
+    def relationship_to_principal_for_group(
```

Comment:
Not something to do now, because I think this ship has sailed anyway, but I wonder if we should have a service object layer that handles this type of thing for the models. Instead of conflating the data model logic with relationship construction, if we had a service object/class to do some of that to interface with the migrator, it might make things more clear, keep good separation, and maybe even make testing easier? Again, not necessarily in the scope of this PR but it's a thought I've had in terms of how we're organizing things.

Reply:
Yeah I could definitely see that. We should talk about how we want to organize logic going forward with the V2 API – it's probably a good chance to get a fresh take and establish a repeatable pattern for some of this stuff. I'm open to moving this somewhere else in the meantime too (top level methods?). Here was at least "close at hand" for the few contexts that needed to deal with group updates currently.

Reply:
Yeah, agreed. As long as we're confident in our tests, we should be able to refactor. 😄